### PR TITLE
Implement ML Info Api

### PR DESCRIPTION
**Pull request by codebrain**
Implemented referencing the YAML tests and server code.

Tests are asserted against the integration test configured cluster.

Requires porting to `master`.

### CODE REVIEW

**Comment by Mpdreamz** [comment]
@codebrain mind retargetting this against `6.6` ?

File: src/Nest/XPack/MachineLearning/MlInfo/ElasticClient-MlInfo.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+using Elasticsearch.Net;
+
+namespace Nest
+{
+	public partial interface IElasticClient
+	{
+		/// <summary>
+		/// Retrieves machine learning job results for one or more categories.
+		/// </summary>
+		IMlInfoResponse MlInfo(Func<MlInfoDescriptor, IMlInfoRequest> selector = null);
```
Comment (russcam) [inline src/Nest/XPack/MachineLearning/MlInfo/ElasticClient-MlInfo.cs]: My vote would be to use `MachineLearningInfo` here

Comment (codebrain) [reply src/Nest/XPack/MachineLearning/MlInfo/ElasticClient-MlInfo.cs]: I had also thought the same thing, however all of the other ML APIs use the `ML` prefix, so we would be breaking consistency. My vote would be to keep as is, can we get a tie breaking vote from @mpdreamz?

Comment (russcam) [reply src/Nest/XPack/MachineLearning/MlInfo/ElasticClient-MlInfo.cs]: The ML prefix is only in the specs though, not in the client source?

Comment (Mpdreamz) [reply src/Nest/XPack/MachineLearning/MlInfo/ElasticClient-MlInfo.cs]: We do have vague plans to move all of the ml api's behind a `MachineLearning` property in 7.0 as well.

I'd prefer `MachineLearningInfo()` for 6.x too.

Comment (codebrain) [reply src/Nest/XPack/MachineLearning/MlInfo/ElasticClient-MlInfo.cs]: Done, renamed as voted! Thanks!

**Comment by codebrain** [comment]
Ported to `master`

**Pull request merged**